scalableminds/webknossos
# Add tags for datasets

Adding a tag system to Datasets, analogous to what Annotations already have.

### URL of deployed dev instance (used for testing):
- https://datasettags.webknossos.xyz

### Backend Changes
- Dataset json now contains additional field `tags`  (list of string)
- Dataset update route now expects `tags` as well (it being missing will be interpreted as empty list), changes tags in database

### Steps to test:
- Open the dataset table. There should be a new row for tags.
- Try adding some tags for a few datasets.
- Clicking on one of the tags should add the tag to the search and hide all datasets that do not have this tag.
- The clicked tag should appear on the top left of the dataset table.
- Clicking on close on the list of tags the datasets are filtered for, should remove that tag from the filtering
- Reloading the page should preserve the filtered tags.
- Adding or removing a tag from a dataset, then waiting a second for the backend call and then reloading the whole page, should show the dataset with the added/removed tag. 
- Reloading a dataset should also not change the displayed tags of a dataset.

### Issues:
- fixes #5829 

------
